ppAnalysispySAL_notebook.ipynb

```diff
@@ -0,0 +1,233 @@
9T4FfgbMYIn3Eo7MPMLDEw8NnxH+OBwHK4e0\no8GKN4ZSIArFW0bbim3Z0HkDDlYOuLu4M+fYHDpv6kzM+LRwtCZTtFdD7eK1GVJnCADLA5fTu1pv\nniY+pUHJBsw9PpeouCgK5ivI1x+kRTn8ZMcnuHzvQovyLdjYeSNdNndh15VdgKZE1nZc+9ypLoAW\n5VpgW9RWswspgKZcoqHUu6UY1G+QoV7oo1DlRDGPoBSIQvEW4unqya7uu7j24Br9qvfjUsQlbKfb\n8mDMA1ztNffvYrK2ZDSl0RRD5MHvjnxHecfyXLp/iUqFKrE8cDkP4x5yJfIKmz/ebOg/9FEollMt\nSUxJxLuHNwN2DWDtOS3gVI93e9Cpsna82KO0h6HNnmt72DB9g3YCKw44BhSD6+euU7VBVRat0AJp\n3X50WymQPIJSIArFW8p7zu+xv/d+9l7fy5j3xwDg8K0DP3f5GS83L0BTIpZmlkzznGawUg+LCSPy\naSRXH1zF3MQcgE0XN2FuYo5PTx9D/3aWdjRf05xPd37Kjm47GHdgHAtOLgAgITmBJmWacDniMp/X\n/tzQptW6VuRLyofNQxvN+rw1kB/CHoRpG+6oE1h5CaVAFIq3mKpFqvJr31/Z/MdmgxKpubgmHqU8\nGOs+FgDrb6ypW6IuzgWcKWVXihrFajCs7jCSUpIMtiRmJma039ieyKeRBq+/EsmWj7dw4d4F6i2r\nx4CaA5h9dDYjfEZwMOQgW7tsZU6zOWy9vJW5zecaZIr7KI6B3QdqbycBmEG8azw9evYA1BJWXkIp\nEIXiLaesQ1l+++Q3dgXtYljdYZS2L03/Xf0Jiw0z7H+4/ehGNadqRMVFcSD4AB0rdWRl+5WGPlJd\npPTc2pMNFzZwoPcBHsU/YtmZZdwbdY/85vmZ5D+JiCcRzD0+l+j4aGwsbOj+bne+b/49M36fwfgG\n47XOBPj94YeVpRWVT1fGysKKwqIwMw7PAOB2jFrCyiu8tAIRQpgIIc4IIXbq152FEBeEEMlCiJpZ\ntHEWQvgJIS4KIc4LIYY9Uz5UCHFJL5vxereiUCiyS4kCJfi1768cvnWYpmWa0uGdDqw6uwqf6z58\nWOpDAJaeWUp0fDRmJmYM8xlGj3d7GE5pXYq4RBHrIjR2bcztmNuMPzCe6Y2ns+faHmYcnsGjcY/4\n+oOveRT/yPCZ3X7uRnJKMt2qdmNu87lMPzyd+s71AQhMCmRE7xFc2HGB1eNW0+PDHiw6vYiQhyHa\nEpZyY5IneGlDQiHECLQDdwWklG2FEBXRnDAvAkZJKQMyaVMUKCqlDBRC2KBFO24npbwshPAAJgAt\npZRJQohCUsqITPpQhoQKRS4RHRdNm/VtcLFzobhNcWYdnQWAQCDJOA4XtFzAoDqDCIsJo/ic4ob8\nUwNO4Rfix+yjs7GxsOH6w+tMbDiRKY2mMOfoHEb5jsrQT9xXcYQ+CqXcvHJYm1vzH4//MNp3NADe\n3b1pWb4l46eMx9rDmsC7gfit8ePYimNULFQRRdbkhiHhSykQIYQzsAKYBoyUUrZNV3YQ+DIzBZJJ\nP9uBeVLKA0KIjcAiKaXfC9ooBaJQ5CJPEp/QeVNnzE3NcXdxZ+z+sVnWjRwTiYOVAzHxMRSYUQCA\nErYlCB0Zyq83fqXrlq6EPw4HYPh7w5FSEpsQS9CDIALCAniSqLmer160Oq3Kt8LNyY3hPsOp51yP\n7Ze3AzCk4BBW/rySRV8sYtieYUScj2DN6DX07Ngzh5+EcZOXFMhmNOVhh6YsXlmBCCFKA/5AVSll\nrBDiDLAD+AjNdGi0lPJUJu2UAlEocpmE5AR6b+vN/Sf36VG1B/139c+ybtxXcViaWRL5JJJC3xUC\nYES9EcxpPoewmDDcl7sTEhViqH958GXKOpRlot9EVp5dyd3YuwA0LdOUjZ03sj94P4N3D+bx6cc8\nCXiiRSn0BItdFiTfSya5TjLlZXnM75kzrP8wPvvksxx9FsZKnnBlIoRoBYRLKQNJC/nySujLV1uA\n4VLKWD3bDCgopawHjEGLBKBQKPIAFqYWrO24lrIFy7IkYAmbOm+igGWBTOsWmFGA4IfBOOZ35PAn\nhwH4/tj3jNs/jiLWRbgy5Aql7NJ8Vo0/MJ7klGRmNJnB4taLDfm+wb44futITEIM81rM48k7T6Au\n5DPNBwJs7WyxL2EPbhCXEMfksZMZ2Hdgzj4IxV/ywhmIEOIbwAsttIsVYAtslVL21sv/cgYihDBD\nC3C5R0r5Q7r83cBMKeWv+vU14D0pZeQz7eWkSZMM1x4eHnh4eLzibSoUiuwgpWSM7xh8rvsws8lM\n+u/s/5xLd4DC+QuzqPUiOlTqwAifEcw9rh3LbVa2Ges7rcfBysFgmAjabGNr160A2E63pWC+gjyM\n09yVFLMpRkm7kniU9mDmTzPhOpSyL8W9R/eQjyVlSpbhVvQtVoxaQac2nXLhKRgH/v7++Pv7G64n\nT56cN5awDJWF+JDMl7BGSSlPZ9FmFRAhpRz5TP5AoISUcpIQogLgK6V8zrWmWsJSKN4sUkqmH57O\n8jPLWdR6EZ95f8b1h8/HIi9tX5p2FdvxbdNvsZyqBa1yd3HnTswdFrdZzEdrPiJZJhvq13OuR7My\nzbgUcYnVHVYzat8o5p+cTwnbEni5efFT4E/EHo3lcf7HUBY63eqEv78/4UHhbN+9nashVxk3bFyu\nPQdjI08sYWWFEKK9EOIWUA/wFkLs0fOLCSG89bQ70BPw1I8ABwghPtK7WAGUEUKcB9YBvV/nRhQK\nRc4ghGBCwwmMqDeCvjv6srTtUmoUrfFcvRtRNwh+GEzDFQ0NS1m/3/odLzcvmq5uiokw4Xj/49hY\n2ODp6smx0GNMOTSFPtX6YGlmybyW89jQaQPR8dEsO7OMdZ3W0a9/P80nloAjkUeIdIhk3ORxdGrT
```

Comment:
Nice job with the G Function.  Here is an example (using R) with the critical bounding envelope added.  http://4.bp.blogspot.com/-OhBzDT4TSZg/VVdEfKo8YQI/AAAAAAAAQ9o/84HOh0_UFHc/s1600/Fig10.tiff